platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java

```diff
@@ -63,15 +64,15 @@ public class DefaultPluginRequestApplicator implements PluginRequestApplicator {
     private final PluginApplicationListener pluginApplicationListenerBroadcaster;
 
     public DefaultPluginRequestApplicator(
-        PluginRegistry pluginRegistry,
+        CorePluginRegistryProvider corePluginRegistryProvider,
         PluginResolverFactory pluginResolverFactory,
         PluginArtifactRepositoriesProvider pluginRepositoriesProvider,
         PluginResolutionStrategyInternal pluginResolutionStrategy,
         PluginInspector pluginInspector,
         PluginVersionTracker pluginVersionTracker,
         ListenerManager listenerManager
     ) {
-        this.pluginRegistry = pluginRegistry;
+        this.corePluginRegistry = corePluginRegistryProvider.getCorePluginRegistry();
```

Comment:
💅 It feels odd that the constructor must extract something from this one argument.